api/src/main/java/org/openmrs/module/fhirExtension/service/ObsBasedDiagnosticReportService.java

```diff
@@ -67,7 +71,8 @@ public List<DiagnosticReport> get(@Nonnull Collection<String> collection) {
 	public DiagnosticReport create(@Nonnull DiagnosticReport diagnosticReport) {
 		try {
 			FhirDiagnosticReport fhirDiagnosticReport = diagnosticReportObsTranslatorImpl.toOpenmrsType(diagnosticReport);
-			validator.validate(fhirDiagnosticReport);
+			diagnosticReportObsValidator.validate(fhirDiagnosticReport);
+			diagnosticReportBasedOnValidator.validate(fhirDiagnosticReport, diagnosticReport);
```

Comment:
no need to penetrate diagnosticReport into validator
```
if(diagnosticReport.hasBasedOn())
     diagnosticReportBasedOnValidator.validate(fhirDiagnosticReport);
```

Reply:
Updated